Week 5/PalindromeLinkedList.cpp

```diff
@@ -0,0 +1,66 @@
+#include <iostream>
+
+using namespace std;
+
+struct ListNode{
+    int val;
+    ListNode *next;
+    ListNode(int x) : val(x), next(NULL) {};
+};
+
+// Reverse function
+ListNode* reverse(ListNode *head) {
+    ListNode *prev = NULL;
+    ListNode *curr = head;
+    ListNode *nxt;
+    while(curr != NULL) {
+        nxt = curr->next;
+        curr->next = prev;
+        prev = curr;
+        curr = nxt;
+    }
+    
+    return prev;
+}
+
+bool isPalindrome(ListNode *head) {
+    // Check for NULL
+    if(head != NULL || head->next != NULL) {
+        throw invalid_argument("NULL VALUE");
+    }
+    
+    // Check is only 1 element
```

Comment:
I really like the comments in your code. Very helpful for others who review your code!